rpc/daemon.go

```diff
@@ -0,0 +1,70 @@
+package rpc
+
+import (
+	"net/http"
+	"runtime/debug"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+)
+
+// DaemonService handles general daemon operations
+type DaemonService struct {
+	server *Server
+	cfg    *Config
+}
+
+// NewDaemonService ...
+func NewDaemonService(server *Server, cfg *Config) *DaemonService {
+	return &DaemonService{
+		server,
+		cfg,
+	}
+}
+
+// ShutdownRequest ...
+type ShutdownRequest struct{}
+
+// ShutdownResponse ...
+type ShutdownResponse struct{}
+
+// Shutdown swapd
+func (s *DaemonService) Shutdown(_ *http.Request, req *ShutdownRequest, resp *ShutdownResponse) error {
+	return s.server.Stop()
```

Comment:
The method here is probably the simplest way for this section of the code to shutdown `swapd`, but just so you know your options, it is not the only way. If `DaemonService` had access to the cancel function of the outermost context, canceling it will also shut everything down.

That is how our signal handler shuts everything down:
https://github.com/AthanorLabs/atomic-swap/blob/master/cmd/swapd/signal_handler.go#L28